packages/hash/api/src/graphql/typeDefs/ontology/property-type.typedef.ts

```diff
@@ -0,0 +1,48 @@
+import { gql } from "apollo-server-express";
+
+export const propertyTypeTypedef = gql`
+  scalar PropertyType
+
+  type IdentifiedPropertyType {
+    propertyTypeVersionedUri: String!
+    createdBy: ID!
+    schema: PropertyType!
+    # TODO: we might need something like
+    # "referencedDataTypes: [DataType!]"
+    # for us to retrieve all referenced data types, and one for referenced property types as well.
+    # This will be helpful for displaying the entire property type in the frontend
+  }
+
+  extend type Query {
+    """
+    Get all property types at their latest version.
+    """
+    getAllLatestPropertyTypes: [IdentifiedPropertyType!]!
+
+    """
+    Get a property type by its versioned URI.
+    """
+    getPropertyType(propertyTypeVersionedUri: String!): IdentifiedPropertyType!
+  }
+
+  extend type Mutation {
+    """
+    Create a property type.
+
+    accountId refers to the account to create the property type in.
+    """
+    createPropertyType(
+      accountId: ID!
+      propertyType: PropertyType!
+    ): IdentifiedPropertyType!
```

Comment:
Nit: documentation for fields should be directly above the field's type definition
```suggestion
  extend type Mutation {
    """
    Create a property type.
    """
    createPropertyType(
      """
      The account to create the property type in.
      """
      accountId: ID!
      propertyType: PropertyType!
    ): IdentifiedPropertyType!
```

(this applies to other places as well)

Reply:
Thanks! Addressed in e4eb4917dbffbe75488572f5aaddece7e6cb273e :)